diffkemp/config.py

```diff
@@ -41,3 +61,60 @@ def __init__(self, snapshot_first, snapshot_second, show_diff,
                                        with -DBUILD_LLREVE=ON")
         elif semdiff_tool is not None:
             raise ConfigException("Unsupported semantic diff tool")
+
+    @classmethod
+    def from_args(cls, args):
+        """
+        Create the configuration from command line arguments.
+        :param args: Command line arguments
+        """
+        # Parse both the new and the old snapshot.
+        snapshot_first = Snapshot.load_from_dir(args.snapshot_dir_old)
+        snapshot_second = Snapshot.load_from_dir(args.snapshot_dir_new)
+
+        # Check if snapshot LLVM versions are compatible with
+        # the current version.
+        llvm_version = get_llvm_version()
+
+        if llvm_version != snapshot_first.llvm_version:
+            sys.stderr.write(
+                "Error: old snapshot was built with LLVM {}, "
+                "current version is LLVM {}.\n".format(
+                    snapshot_first.llvm_version, llvm_version
+                )
+            )
+            sys.exit(errno.EINVAL)
+
+        if llvm_version != snapshot_second.llvm_version:
+            sys.stderr.write(
+                "Error: new snapshot was built with LLVM {}, "
+                "current version is LLVM {}.\n".format(
+                    snapshot_second.llvm_version, llvm_version
+                )
+            )
+            sys.exit(errno.EINVAL)
```

Comment:
Could we move this code into `Snapshot.load_from_dir`? It would simplify this method and we could avoid the duplication here.